### PR TITLE
fix(refs T19759): fix warnings/errors [vue 3 upgrade]

### PR DESCRIPTION
**Pull request by hwiem**
**Integration PR:** #1151
**Ticket:** https://yaits.demos-deutschland.de/T19759

Fixes some console warnings/errors.

- `config.productionTip` has been removed in vue 3
see https://v3-migration.vuejs.org/breaking-changes/global-api.html#a-new-global-api-createapp
- `app.config.compilerOptions.whitespace` needs to be defined
see https://vuejs.org/api/application.html#app-config-compileroptions-whitespace
- `Vue.set()` has been removed and is no longer needed
- __VUE_OPTIONS_API__ and __VUE_PROD_DEVTOOLS__ need to be defined
see http://link.vuejs.org/feature-flags
- `Vue.prototype` is no longer available in vue 3. Instead, `app.config.globalProperties` should be used
see https://v3-migration.vuejs.org/breaking-changes/global-api.html#a-new-global-api-createapp
- hooks in directives have changed:
bind -> beforeMount
inserted -> mounted
update -> updated
componentUpdated -> updated
unbind -> unmounted
see https://v3-migration.vuejs.org/breaking-changes/custom-directives.html#_3-x-syntax
- `beforeDestroy` hook has been renamed to `beforeUnmount`
- use deep watchers for arrays, see https://v3-migration.vuejs.org/breaking-changes/watch.html

### Related PRs

- [x] https://github.com/demos-europe/demosplan-core/pull/2133 (Needs to be merged first)

### PR Checklist

- [ ] Tests updated/created // Tests are ignored for now
- [x] Link all relevant tickets

### CODE REVIEW

File: client/js/components/statement/assessmentTable/AssessmentTableFilter.vue [hunk 1]
```diff
@@ -60,8 +60,7 @@
       class="c-at__controls o-sticky o-sticky--border space-stack-xs u-pt-0_5 u-pb-0_25"
       ref="header">
       <search-and-sorting
-        :search-term="searchTerm"
-        :has-changed-statements="hasChangedStatements" />
```
Comment (salisdemos) [inline client/js/components/statement/assessmentTable/AssessmentTableFilter.vue:64]: why is this obsolet?

Comment (hwiem) [reply client/js/components/statement/assessmentTable/AssessmentTableFilter.vue:64]: SearchAndSorting.vue doesn't have a `hasChangedStatements` prop, so it was added as an attribute. I could not
find any usage of it, so I believe that it was added here accidentally and can be removed

File: yarn.lock [hunk 2]
```diff
@@ -1385,9 +1385,9 @@
   resolved "https://registry.yarnpkg.com/@csstools/selector-specificity/-/selector-specificity-3.0.0.tgz#798622546b63847e82389e473fd67f2707d82247"
   integrity sha512-hBI9tfBtuPIi885ZsZ32IMEU/5nlZH/KOVYJCOh7gyMxaVLGmLedYqFN6Ui1LXkI8JlC8IsuC0rF0btcRZKd5g==
 
-"@demos-europe/demosplan-ui@github:demos-europe/demosplan-ui#f_set_demosplan-ui_invue3_compat_mode":
+"@demos-europe/demosplan-ui@github:demos-europe/demosplan-ui#i_vue3_migration":
```
Comment (salisdemos) [inline yarn.lock:1388]: I assume, that should not be part of the PR

**Review by salisdemos — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 2 above)
Comment (hwiem) [reply yarn.lock:1388]: Shouldn't it? The branch `f_set_demosplan-ui_invue3_compat_mode` doesn't exist anymore

Comment (salisdemos) [reply yarn.lock:1388]: never mind
